gtk4-macros/src/blueprint.rs

```diff
@@ -1,28 +1,36 @@
 // Take a look at the license at the top of the repository in the LICENSE file.
 
 use std::{
-    io::{Error, ErrorKind, Read, Result, Write},
+    io::Write,
     process::{Command, Stdio},
 };
 
-pub(crate) fn compile_blueprint(blueprint: &[u8]) -> Result<String> {
+pub(crate) fn compile_blueprint(blueprint: &[u8]) -> Result<String, String> {
     let mut compiler = Command::new("blueprint-compiler")
         .args(["compile", "-"])
         .stdin(Stdio::piped())
         .stdout(Stdio::piped())
         .spawn()
-        .unwrap_or_else(|_| panic!("blueprint-compiler not found"));
-
+        .map_err(|e| format!("blueprint-compiler couldn't be spawned: {e}"))?;
     let mut stdin = compiler.stdin.take().unwrap();
-    stdin.write_all(b"using Gtk 4.0;\n")?;
-    stdin.write_all(blueprint)?;
+    if let Err(e) = stdin
+        .write_all(b"using Gtk 4.0;\n")
```

Comment:
@bilelmoussaoui This still inserts the line for backwards compat. Your call but removing this seems like it would break existing code

Reply:
i wouldn't remove it, no